### PR TITLE
Exam updates

### PR DESCRIPTION
**Pull request by christianmemije**
### Summary

Integrates cards into exam creation.
Updated 4 strings. 

TODO: Add gherkin stories.
…

### Reviewer guidance

Test that everything works. I guess might require gherkin stories.

…

----

### Contributor Checklist

- [ ] Contributor has fully tested the PR manually
- [ ] PR has the correct target branch and milestone
- [ ] PR has 'needs review' or 'work-in-progress' label
- [ ] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label

### Reviewer Checklist

- [ ] Automated test coverage is satisfactory
- [ ] Reviewer has fully tested the PR manually
- [ ] PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- [ ] External dependencies files were updated (`yarn` and `pip`)
- [ ] Documentation is updated
- [ ] Link to diff of internal dependency change is included
- [ ] CHANGELOG.rst is updated for high-level changes
- [ ] Contributor is in AUTHORS.rst

### CODE REVIEW

File: kolibri/plugins/coach/assets/src/views/exams/CreateExamPage/index.vue [hunk 1]
```diff
@@ -192,49 +181,119 @@
...
       noneSelected: 'No exercises are selected',
-      searchContent: 'Search for content within channel',
       preview: 'Preview',
       saveButtonlabel: 'Save',
       added: 'Added',
       removed: 'Removed',
       selected: '{count, number, integer} total selected',
-      name: 'Name',
       documentTitle: 'Create new exam',
+      exitSearchButtonLabel: 'Exit search',
```
Comment (christianmemije) [inline kolibri/plugins/coach/assets/src/views/exams/CreateExamPage/index.vue:187]: New string.

File: kolibri/plugins/coach/assets/src/views/lessons/LessonContentPreviewPage/SelectOptions.vue [hunk 2]
```diff
@@ -39,34 +39,14 @@
     },
     $trs: {
       removeLabel: 'Remove',
-      addToLessonButtonLabel: 'Add to lesson',
-      addedToLessonIndicator: 'Added to lesson',
+      addToLessonButtonLabel: 'Add',
```
Comment (christianmemije) [inline kolibri/plugins/coach/assets/src/views/lessons/LessonContentPreviewPage/SelectOptions.vue]: New string.

File: kolibri/plugins/coach/assets/src/views/lessons/LessonContentPreviewPage/SelectOptions.vue [hunk 3]
```diff
@@ -39,34 +39,14 @@
...
+      addedToLessonIndicator: 'Added',
```
Comment (christianmemije) [inline kolibri/plugins/coach/assets/src/views/lessons/LessonContentPreviewPage/SelectOptions.vue]: New string.

File: kolibri/core/content/api.py [hunk 4]
```diff
@@ -210,6 +211,11 @@ class OptionalPageNumberPagination(pagination.PageNumberPagination):
     page_size_query_param = "page_size"
 
 
+class SQSum(Subquery):
+    template = "(SELECT SUM(\"%(field)s\") FROM (%(subquery)s))"
```
Comment (ralphiee22) [inline kolibri/core/content/api.py]: Saw that you can also pass in a kwarg of template but I do think this way is cleaner!

**Review by ralphiee22 — COMMENTED** [review]
Extensive backend tests 👍

File: kolibri/plugins/coach/assets/src/modules/examCreation/actions.js [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+import router from 'kolibri.coreVue.router';
+import { ContentNodeKinds } from 'kolibri.coreVue.vuex.constants';
+import { PageNames } from '../../constants';
+import { _createExam } from '../shared/exams';
+
+const snackbarTranslator = createTranslator('ExamCreateSnackbarTexts', {
+  newExamCreated: 'New exam created',
+});
+
+export function resetExamCreationState(store) {
```
Comment (jonboiser) [inline kolibri/plugins/coach/assets/src/modules/examCreation/actions.js:17]: If these one-mutation type actions aren't used in too many components, I like using the mapMutations helper in the components, e.g. `...mapMutations('examCreation', { setTitle: 'SET_TITLE' })`

File: kolibri/plugins/coach/assets/src/views/exams/CreateExamPage/index.vue [hunk 6]
```diff
@@ -436,6 +447,27 @@
           );
         }
       },
+      contentIsIndeterminate(content) {
+        if (content.kind === ContentNodeKinds.TOPIC) {
+          const everyExerciseSelected = content.exercises.every(
```
Comment (jonboiser) [inline kolibri/plugins/coach/assets/src/views/exams/CreateExamPage/index.vue:457]: Using similar code, you could also filter and count the nodes in `content.exercises` that are in `this.selectedExercises`. Then you can determine the checkbox state on whether the length of that filtered list is 0 (not checked), equal to `content.exercises.length` (checked), or less than `content.exercises.length` (indeterminate)

Comment (jonboiser) [reply kolibri/plugins/coach/assets/src/views/exams/CreateExamPage/index.vue:457]: For this kind of scenario, I sometimes like to use lodash/xor to get the difference between sets. e.g. if selected - content.exercises is empty, then checked, etc.

**Review by jonboiser — APPROVED** [review]
Looks good!

**Comment by radinamatic** [comment]
Wow, that's a Bible, not a Gherkin... :joy:

**Pull request merged**